GiniVision/Classes/Core/CameraViewController.swift

```diff
@@ -779,34 +821,44 @@ extension CameraViewController {
-        alertViewController.addAction(UIAlertAction(title: "Zugriff erteilen", style: .default, handler: { _ in
-            alertViewController.dismiss(animated: true, completion: nil)
-            UIApplication.shared.openAppSettings()
-        }))
-        
-        self.present(alertViewController, animated: true, completion: nil)
-    }
-    
-    fileprivate func showNotValidDocumentError(message: String) {
-        
-        let alertViewController = UIAlertController(title: nil, message: message, preferredStyle: .alert)
-        alertViewController.addAction(UIAlertAction(title: "Abbrechen", style: .cancel, handler: { _ in
-            alertViewController.dismiss(animated: true, completion: nil)
-        }))
-        alertViewController.addAction(UIAlertAction(title: "Andere Datei wählen", style: .default, handler: { _ in
-            self.showImportFileSheet()
-        }))
+    func showErrorDialog(for error: Error, possitiveAction: (() -> Void)? = nil) {
+        let message: String
+        var cancelActionTitle: String = "Abbrechen"
+        var confirmActionTitle: String? = "Andere Datei wählen"
+        var confirmAction: (() -> Void)?
+        
+        switch error {
+        case let validationError as DocumentValidationError:
+            message = validationError.message
+            confirmAction = self.showImportFileSheet
+        case let customValidationError as CustomDocumentValidationError:
+            message = customValidationError.message
+            confirmAction = self.showImportFileSheet
+        case let pickerError as FilePickerError:
+            message = pickerError.message
+            if pickerError == .filesPickedCountExceeded {
+                confirmActionTitle = "Seitenübersicht"
+                confirmAction = { [weak self] in
+                    guard let `self` = self else { return }
+                    self.delegate?.cameraDidTapMultipageReviewButton(self)
+                }
+            } else if pickerError == .mixedDocumentsUnsupported {
+                cancelActionTitle = NSLocalizedStringPreferred("ginivision.camera.mixedarrayspopup.cancel",
+                                                        comment: "cancel button text for popup")
+                confirmActionTitle = NSLocalizedStringPreferred("ginivision.camera.mixedarrayspopup.usePhotos",
+                                                           comment: "use photos button text in popup")
+            }
+
+        default:
+            message = DocumentValidationError.unknown.message
+        }
+
+        let dialog = errorDialog(withMessage: message,
+                                 cancelActionTitle: cancelActionTitle,
+                                 confirmActionTitle: confirmActionTitle,
+                                 confirmAction: possitiveAction ?? confirmAction)
         
-        present(alertViewController, animated: true, completion: nil)
+        present(dialog, animated: true, completion: nil)
     }
 }
 
```

Comment:
Line 864... Please split it or something.

Reply:
#164